routing/router.go

```diff
@@ -1677,8 +1677,16 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 
 	// We'll continue until either our payment succeeds, or we encounter a
 	// critical error during path finding.
-	var lastError error
-	for {
+	var (
+		lastError error
+		route     *route.Route
+		paymentID uint64
+	)
+
+	// sendNewAttempt is a helper method that creates and sends a new
+	// payment to the switch. We use this when no payment has been sent to
+	// the switch already, or the previous attempt failed.
+	sendNewAttempt := func() error {
```

Comment:
Why a function closure vs moving it into a distinct method?

Reply:
No longer applicable.